src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -400,6 +411,23 @@ public HttpResponse doFinishLogin(StaplerRequest request)
         return HttpResponses.redirectToContextRoot();   // referer should be always there, but be defensive
     }
 
+    private void storeCredentials(String accessToken, GithubAuthenticationToken auth) throws IOException {
+        CredentialsStore credentialsStore = CredentialsProvider.lookupStores(Jenkins.getInstance()).iterator().next();
+        Credentials github = Iterables.find(credentialsStore.getCredentials(Domain.global()), new Predicate<Credentials>() {
```

Comment:
What is wrong with just using the lookup method and the credentials matcher API (we have the CQL query language specifically so that the query can be converted to the external provider's query format and thus reducing the number of credentials that need (partial) fetching from the external provider)